airflow/jobs/scheduler_job_runner.py

```diff
@@ -1901,6 +1901,7 @@ def _orphan_unreferenced_datasets(self, session: Session = NEW_SESSION) -> None:
                 isouter=True,
             )
             .group_by(DatasetModel.id)
+            .where(~DatasetModel.is_orphaned)
```

Comment:
Not too familiar with this area, but is it possible to create/update an existing unit test to check for this condition? (maybe as part of `tests/jobs/test_scheduler_job.py::TestSchedulerJob::test_dataset_orphaning`)

Reply:
Hi @shahar1,

Thansk for the feedback!
I have thought about creating/updating a test but I believe it does not make sense.
There is currently this test here: https://github.com/apache/airflow/blob/63662044583031fc27d98af02f2913d324245db0/tests/jobs/test_scheduler_job.py#L5639

The logic used to be as follows if we have:
- dataset 1 (already orphaned)
- dataset 2 (not orphaned, but has no more references)
- dataset 3 (not orphaned, has references)

The result before my change would be:
- dataset 1 (orphaned, but set to orphaned again)
- dataset 2 (orphaned)
- dataset 3 (not orphaned)

The result after my change would be:
- dataset 1 (orphaned, but not updated)
- dataset 2 (orphaned)
- dataset 3 (not orphaned)


So in the end the result is the same, the only think I can maybe check is if the updatedAt field of dataset 1 has changed.

Reply:
I'd be happy if you could give it a shot :)

Reply:
@shahar1 I have updated the PR with a test, the update_at field was indeed changed in the previous implementation :) I tested it by commenting the line changed, and then the new test fails

Reply:
Thanks for creating the test!
Currently it seems that all tests pass (the one that fails is unrelated) - so I think that we're good to go.